### DISCUSSION TITLE
Dark mode on Windows 11 in Toga

### DISCUSSION THREAD

**Discussion by gryznar** [beeware/toga#1957] [Q&A]
Hello. I am new to Toga and maybe I am missing something, but despite the fact that dark mode is set in windows 11, app window is rendered in light mode. If am I doing sth wrong, could I get some tip or advice?

**Comment by freakboy3742** [comment]
To the best of my knowledge, we're not doing anything to explicitly *disable* dark mode; however, we're also not be doing anything to *enable* dark mode. I can't say I know what is required; if you're able to work out what Winforms APIs enable dark mode support, we're happy to accept those as a PR to the Winforms backend.

**Comment by gryznar** [comment]
But in analogical discussion for iOS it is said, that it is enough to set dark mode on system level. Won't it be applied to Windows?

**Reply by freakboy3742** [reply]
It seems weird to have to say this, but... iOS isn't Windows. They use *completely* different APIs. If a feature is implemented on the iOS backend, that doesn't mean it's implemented on Windows, or vice versa. If a bug exists on the iOS backend, that doesn't mean the same bug exists on the Windows backend, or vice versa.

On iOS, Dark mode "just happens". There's no explicit API required to turn it on. Evidently that isn't true on Winforms. I don't know what the required API *is*, but it evidently exists, because, as I said, we're not explicitly *disabling* dark mode support.

**Reply by gryznar** [reply]
I see. I have supposed, that this can happen automatically ;)

**Reply by freakboy3742** [reply]
I've just done a quick search, and it would appear that Winforms doesn't support Dark mode *at all*: 
* https://github.com/dotnet/winforms/issues/7641
* https://github.com/dotnet/winforms/issues/6270

**Comment by yohasa** [comment]
private void MainForm_Load(object sender, EventArgs e)
{
    SetDarkMode();
}

private void SetDarkMode()
{
    // Modifier ici les couleurs pour le mode sombre
    // Exemple : modifier la couleur de fond du formulaire
    this.BackColor = Color.FromArgb(31, 31, 31);
    // Modifier d'autres contrôles, tels que les boutons, les étiquettes, etc.

    // Si vous souhaitez que les contrôles internes suivent automatiquement le thème sombre,
    // vous pouvez utiliser la propriété 'EnableVisualStyles' du gestionnaire d'applications.
    // Application.EnableVisualStyles();
}
